dig_test.go

```diff
@@ -3566,3 +3567,91 @@ func TestEndToEndSuccessWithAliases(t *testing.T) {
 	})
 
 }
+
+func TestConcurrency(t *testing.T) {
+	// Ensures providers will run at the same time
+	t.Run("TestMaxConcurrency", func(t *testing.T) {
+		t.Parallel()
+
+		type (
+			A int
+			B int
+			C int
+		)
+
+		var (
+			timer           = time.NewTimer(10 * time.Second)
+			max       int32 = 3
```

Comment:
minor: We can define a `const Max = 3`; it'll have type "untyped int" and won't need to be cast to int32.

Reply:
Thanks, that's a language feature I tend to forget about!